src/tag.rs

```diff
@@ -246,12 +317,23 @@ impl Tag {
         source: &mut S,
     ) -> Result<(Self, bool), S::Err> {
         let byte = source.take_u8()?;
-        if (byte & 0x1F) == 0x1F {
-            // If all five lower bits are 1, the tag is encoded in multiple
-            // bytes. We don’t support that.
-            xerr!(return Err(decode::Error::Unimplemented.into()))
+        // clear constructed bit
+        let mut data = [byte & !Tag::CONSTRUCTED_MASK, 0, 0, 0];
+        if (data[0] & Tag::SINGLEBYTE_DATA_MASK) == Tag::SINGLEBYTE_DATA_MASK {
+            let mut i = 1;
```

Comment:
I think I would prefer this as a `for i in 0..3` loop with a early return on success and an error return on finishing the loop. Seems more robust.

Reply:
I tried to do it that first try, but it did not go smoothly since I had something like this
```rust
    pub fn take_from<S: decode::Source>(
        source: &mut S,
    ) -> Result<(Self, bool), S::Err> {
        let byte = source.take_u8()?;
        // clear constructed bit
        let mut data = [byte & !Tag::CONSTRUCTED_MASK, 0, 0, 0];
        if (data[0] & Tag::SINGLEBYTE_DATA_MASK) == Tag::SINGLEBYTE_DATA_MASK {
            for i in 1..=3 {
                data[i] = source.take_u8()?;
                if data[i] & Tag::LAST_OCTET_MASK == 0 {
                    return Ok((Tag(data), byte & Tag::CONSTRUCTED_MASK != 0));
                }
            }
            return xerr!(return Err(decode::Error::Unimplemented.into()));
        }
        Ok((Tag(data), byte & Tag::CONSTRUCTED_MASK != 0))
    }
```
But I could not find a clean way to put the first byte case into the loop.

Reply:
This looks fine to me, since the first byte is special anyway.

I’d probably move the final Ok into an else branch for the if so it wouldn’t need the return for the Err below the for loop.

Reply:
Changed.